analytics/analytics.py

```
@@ -86,14 +317,31 @@ def _refresh_headers(self):
             "Content-Type": "application/json",
         }
 
-    def submit(self, txn: AnalyticsTxn):
+    def submit_transaction(self, txn: AnalyticsTxn):
+        # TODO: We should make this fail safely, as to not interrupt the everyday
+        # business logic products do.
+        # However, given that we do not have appropriate logging infrastructure
```

What's the appropriate infra we'd need to change here? Agree on not failing silently rn

---

We should be surrounding this with a try catch and logging on failure so we can "silently" fail but be notified its failing. 

Think about what we would want if the analytics server went down for example; we wouldn't want all penn labs products routes to start failing because this function fails